src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -455,30 +455,55 @@ class GoogleExtensions(
       key <- googleKeyCache.getKey(pet)
     } yield key
 
-  def getPetServiceAccountToken(user: SamUser, project: GoogleProject, scopes: Set[String], samRequestContext: SamRequestContext): Future[String] =
-    getPetServiceAccountKey(user, project, samRequestContext).unsafeToFuture().flatMap { key =>
-      getAccessTokenUsingJson(key, scopes)
+  def getPetServiceAccountToken(user: SamUser, project: GoogleProject, scopes: Set[String], samRequestContext: SamRequestContext): IO[String] =
+    getPetServiceAccountKey(user, project, samRequestContext).flatMap { key =>
+      IO.fromFuture(IO(getAccessTokenUsingJson(key, scopes)))
     }
 
+  def getPetServiceAccountToken(
```

Comment:
The general pattern is we overload these methods: one with a `SamUser` and one with a `WorkbenchEmail`. So now we have 8 total combinations:
```
getPetServiceAccountKey(user|email)
getPetServiceAccountToken(user|email)
getArbitraryServiceAccountKey(user|email)
getArbitraryServiceAccountToken(user|email)
```